http/src/main/resources/init-resources/gce-init.sh

```diff
@@ -220,9 +220,13 @@ START_TIME=$(date +%s)
 STEP_TIMINGS=($(date +%s))
 
 
-# Use specific docker compose command if the container is coming from GCR, see  https://hub.docker.com/r/cryptopants/docker-compose-gcr
-# TODO - Also check for GAR see https://broadworkbench.atlassian.net/browse/IA-4518
-if grep -qF "gcr.io" <<< "${JUPYTER_DOCKER_IMAGE}${RSTUDIO_DOCKER_IMAGE}${PROXY_DOCKER_IMAGE}${WELDER_DOCKER_IMAGE}" ; then
+# everything in the public broad us.gcr.io bucket has been migrated to GAR
+if grep -qF "us.gcr.io" <<< "${JUPYTER_DOCKER_IMAGE}${RSTUDIO_DOCKER_IMAGE}${PROXY_DOCKER_IMAGE}${WELDER_DOCKER_IMAGE}" ; then
```

Comment:
Might be a silly question sorry, but could there still be private repos out there that also start with this path? I do not thinks so since I am pretty sure the paths are unique but still wanted to double check

Reply:
Also you might not need this if, elif, else You could do:

if private and stored in GCR: Authorize GCR
else: Authorize GAR

What do you think?

Reply:
ooh yeah good points...not entirely sure how to determine private vs. public at this level, but I'll think on that